enterprise_gateway/services/processproxies/crd.py

```diff
@@ -40,6 +42,53 @@ async def launch_process(
         await super().launch_process(kernel_cmd, **kwargs)
         return self
 
+    def get_container_status(self, iteration: int | None) -> str:
+        """Determines submitted CRD application status
+
+        Submitting a new kernel application to the spark operator
+        will take a while to be Running and can also fail before it
+        can actually start any pods. So we will validate that the
+        application succeeds on the Spark Operator side and then
+        delegate to parent to check if the application pod is running.
+
+        Returns
+        -------
+        Empty string if the container cannot be found otherwise.
+        The pod application status in case of success on Spark Operator side
+        Or the retrieved spark operator submission status in other cases (e.g. Failed)
+        """
+        with suppress(Exception):
+            custom_resource = client.CustomObjectsApi().get_namespaced_custom_object(
+                self.group,
+                self.version,
+                self.kernel_namespace,
+                self.plural,
+                self.kernel_resource_name,
+            )
+
+            if not custom_resource:
+                return ""
+
+            application_state = custom_resource['status']['applicationState']['state'].lower()
+
+            if iteration:
+                self.log.debug(f"Checking CRD status: {application_state}")
+
+            if application_state in self.get_error_states():
+                exception_text = self._get_exception_text(
+                    custom_resource['status']['applicationState']['errorMessage']
+                )
+                error_message = (
+                    f"CRD submission for kernel {self.kernel_id} failed: {exception_text}"
+                )
+                self.log.debug(error_message)
+            elif application_state == "running" and not self.assigned_host:
```

Comment:
Similar to the previous comment, could we change this to `else:` and call the superclass unconditionally at this point.  It already contains logic to scrutinize the `assigned_host` and `"running"` status.